File: swig/lua/test-r_asm.lua
require "r_asm"

a = r_asm.RAsm ()
a:use ("x86.olly")
opcode = "mov eax, 33"
foo = a:massemble (opcode)
if foo ~= nil then
	print (string.format ('%s =  %s', opcode, foo.buf_hex))
else
	print "Cannot assemble opcode"
end

File: r2-bindings/python/test-r_asm.py
from r_asm import *

def ass(a, arch, op):
	print "---------------------------->8- - - - - -"
	print "OPCODE: %s"%op
	a.use (arch)
	print "ARCH: %s"%arch
	code = a.massemble (op)
	if code is None:
		print "HEX: Cannot assemble opcode"
	else:
		print "HEX: %s"%code.buf_hex
	
a = RAsm()
ass (a, 'x86.olly', 'mov eax, 33')
ass (a, 'java', 'bipush 33')
